db/db_impl/db_impl_files.cc

```diff
@@ -572,9 +566,17 @@ void DBImpl::PurgeObsoleteFiles(JobContext& state, bool schedule_only) {
       case kTableFile:
         // If the second condition is not there, this makes
         // DontDeletePendingOutputs fail
+        // FIXME: but should remove if it came from sst_delete_files?
```

Comment:
Can you clarify what is it to be fixed? Is it we need to remove the file (i.e, !keep) even if number >= state.min_pending_output when the file comes from sst_delete_files??

Reply:
Yes, I believe we should not be keeping it if it came from sst_delete_files. Such files were live in a Version but no longer are, so AFAIK there should be no ambiguity about whether they will appear in a future Version. But needs to be fully evaluated.